controller/invitation_blackbox_test.go

```diff
@@ -293,6 +290,74 @@ func (s *TestInvitationREST) TestAcceptInvitationFailsForNonUUIDCode() {
 	require.NotNil(s.T(), parameters.Get("error"))
 }
 
+func (s *TestInvitationREST) TestRescindInvitation() {
+	team := s.Graph.CreateTeam()
+	invitee := s.Graph.CreateUser()
+	inv := s.Graph.CreateInvitation(team, invitee)
+
+	r := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.IdentityResourceTypeTeam))
+	r.AddScope(authorization.ManageTeamMembersScope)
+	team.AssignRole(&s.testIdentity, r.Role())
+
+	service, controller := s.SecuredController(s.testIdentity)
+
+	response := test.RescindInviteInvitationOK(s.T(), service.Context, service, controller, inv.Invitation().InvitationID.String())
+
+	require.NotNil(s.T(), response.Header().Get("Location"))
+
+	// The invitation should no longer be there after rescinding
+	_, err := s.Application.InvitationRepository().Load(s.Ctx, inv.Invitation().InvitationID)
+	require.Error(s.T(), err)
+	require.IsType(s.T(), errors.NotFoundError{}, err)
+}
+
+func (s *TestInvitationREST) TestRescindInvitationFailsForUnauthorized() {
+	team := s.Graph.CreateTeam()
+	invitee := s.Graph.CreateUser()
+	inv := s.Graph.CreateInvitation(team, invitee)
+
+	service, controller := s.SecuredController(s.testIdentity)
+
+	response, _ := test.RescindInviteInvitationInternalServerError(s.T(), service.Context, service, controller, inv.Invitation().InvitationID.String())
+
+	require.NotNil(s.T(), response.Header().Get("Location"))
+
+	_, err := s.Application.InvitationRepository().Load(s.Ctx, inv.Invitation().InvitationID)
+	require.NoError(s.T(), err)
+}
+
+func (s *TestInvitationREST) TestRescindInvitationFailsForInvalidID() {
+	team := s.Graph.CreateTeam()
+	invitee := s.Graph.CreateUser()
+	s.Graph.CreateInvitation(team, invitee)
+
+	r := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.IdentityResourceTypeTeam))
+	r.AddScope(authorization.ManageTeamMembersScope)
+	team.AssignRole(&s.testIdentity, r.Role())
+
+	service, controller := s.SecuredController(s.testIdentity)
+
+	response, _ := test.RescindInviteInvitationInternalServerError(s.T(), service.Context, service, controller, uuid.NewV4().String())
+
+	require.NotNil(s.T(), response.Header().Get("Location"))
+}
+
+func (s *TestInvitationREST) TestRescindInvitationFailsForNonUUIDValue() {
+	team := s.Graph.CreateTeam()
+	invitee := s.Graph.CreateUser()
+	s.Graph.CreateInvitation(team, invitee)
+
+	r := s.Graph.CreateRole(s.Graph.LoadResourceType(authorization.IdentityResourceTypeTeam))
+	r.AddScope(authorization.ManageTeamMembersScope)
+	team.AssignRole(&s.testIdentity, r.Role())
+
+	service, controller := s.SecuredController(s.testIdentity)
+
+	response, _ := test.RescindInviteInvitationInternalServerError(s.T(), service.Context, service, controller, "foo")
```

Comment:
This doesn't look like a InternalServerError to me :) Should be either a NotFound 404 ( if we don't wanna expose that uuid is our implementation detail ), or a 400 Bad Request,